tests/test_update_file_manager.py

```diff
@@ -71,3 +77,792 @@ def test_check_if_update_based_on_last_modified(database, mocker, mock_db):
+        "https://example.com/test.txt", str(test_file)
+    )
+    mock_db.add_ips_to_IoC.assert_any_call(mocker.ANY)
+    mock_db.add_domains_to_IoC.assert_any_call({"example.com": mocker.ANY})
+    assert result is True
+
+
+def test_parse_ti_feed_invalid_data(mocker, mock_db, tmp_path):
+    """Test parse_ti_feed with invalid data."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    update_manager.url_feeds = {
+        "https://example.com/invalid.txt": {
+            "threat_level": "low",
+            "tags": ["tag3"],
+        }
+    }
+    test_data = """# Comment
+    invalid_data,Description
+    """
+    test_file = tmp_path / "invalid.txt"
+    test_file.write_text(test_data, encoding="utf-8")
+    mocker.patch("builtins.open", mock_open(read_data=test_data))
+    result = update_manager.parse_ti_feed(
+        "https://example.com/invalid.txt", str(test_file)
+    )
+    mock_db.add_ips_to_IoC.assert_not_called()
+    mock_db.add_domains_to_IoC.assert_not_called()
+    assert result is False
+
+
+@pytest.mark.parametrize(
+    "file_content, cached_hash, expected_result",
+    [
+        ("test content", {}, True),  # Testcase1 New file
+        ("new content", "old_hash", True),  # Testcase2 Updated file
+        # Testcase3 Unchanged file
+        (
+            "test content",
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+            True,
+        ),
+    ],
+)
+def test_check_if_update_org(
+    mock_db, tmp_path, file_content, cached_hash, expected_result
+):
+    """Test check_if_update_org with different file and cache scenarios."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    test_file = tmp_path / "test_org"
+    test_file.write_text(file_content, encoding="utf-8")
+    mock_db.get_TI_file_info.return_value = {"hash": cached_hash}
+    result = update_manager.check_if_update_org(str(test_file))
+    assert result is expected_result
+
+
+@pytest.mark.parametrize(
+    "status_code, expected_result, db_call_count",
+    [  # Testcase1:Successful download and update
+        (200, True, 1),
+        # Testcase2:Download failed
+        (404, False, 0),
+        # Testcase3:Server error
+        (500, False, 0),
+    ],
+)
+def test_update_mac_db(
+    mocker, mock_db, status_code, expected_result, db_call_count
+):
+    """
+    Test update_mac_db with different response status codes.
+    """
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    update_manager.mac_db_link = "https://example.com/mac_db.json"
+    mock_response = mocker.Mock()
+    mock_response.status_code = status_code
+
+    mock_response.text = '[{"mac":"00:00:00:00:00:01","vendor":"VendorA"}]'
+    update_manager.responses["mac_db"] = mock_response
+
+    result = update_manager.update_mac_db()
+    assert result is expected_result
+    assert mock_db.set_TI_file_info.call_count == db_call_count
+
```

Comment:
we should mock this one
```py
with open(path_to_mac_db, "w")
```
because when update_mac_db() is called, it write to "databases/macaddress-db.json"